airmozilla/manage/views.py

```diff
@@ -3687,6 +3688,18 @@ def picture_delete(request, id):
 
 
 @staff_required
+@permission_required('main.delete_picture')
+@transaction.commit_on_success
+@json_view
+def picture_delete_all(request):
+    event = get_object_or_404(Event, id=request.POST.get('event'))
+    Picture.objects.filter(event=event).delete()
+    event.picture = None
```

Comment:
This is not necessarily true. 

Suppose an event has a picture associated with it from the general pick of pictures in the gallery. Then, it shouldn't reset the `event.picture`. I.e. we need something like this:
```pyhon
pictures = Picture.objects.filter(event=event)
if event.picture and event.picture in pictures:
    assert event.placeholder_img
    event.picture = None
    event.save()
pictures.delete()
```

See what I mean? Also, it would be great if you could add a dedicated test that tests both paths. One test where the event has a picture from those belonging to the event and one where the event has a picture but it's a general picture not belonging to any event.

Sorry. I understand that this "cross foreign key" situation is confusing.